### PR TITLE
[Enhancement] Disable operator's mem_tracker if profile is disable

### PR DESCRIPTION
**Pull request by liuyehcf**
Signed-off-by: liuyehcf <1559500551@qq.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

#17023 bring performance deduction, the reason comes from two aspects:
1. 3 atomic rather than 1 atomic are updated each time for MemTracker updation.
2. there have been more context switch. when calling `push_chunk` and `pull_chunk`, the thread_local will change to operator's mem_tracker, in memory intensive cases, the total amount of context switch will be tremendous, which will finally lead to MemTrack's updation.

## Experiment

benchmark_04(1fe, 3be, 16c/64G)

```sql
select count(*) from ssb_200g.lineorder_flat where LO_ORDERDATE != '1994-04-12';
```

|  | 1 concurrency | 10 concurrency |
|:--|:--|:--|
| before #17023(disable profile) | 177ms | 0.716s |
| before #17023(enable profile) | 183ms | 0.7.66s |
| #17023(disable profile) | 417ms | 1.596s |
| #17023(enable profile) | 435ms | 1.596s |
| this pr(disable profile) | 176ms | 0.699s |
| this pr(enable profile) | 390ms | 1.317s |

So we can draw the conclusion that this pr's performance is same as before #17023 if profile is disabled. But there still has substantial deduction in performance if profile is enabled for memory intensive situations.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|7|00.00%|[1626, 1627, 1665, 1666, 1667, 1668, 1669]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|7|00.00%|[1626, 1627, 1665, 1666, 1667, 1668, 1669]|

**Review by murphyatwork — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/runtime/current_thread.h [hunk 1]
```diff
@@ -53,20 +53,18 @@ class CurrentThread {
...
-            if (_cache_size > 0) {
-                _allocated_cache_size -= _cache_size;
-            } else {
-                _deallocated_cache_size += _cache_size;
-            }
-            cur_tracker->update_allocation(_allocated_cache_size);
-            cur_tracker->update_deallocation(_deallocated_cache_size);
             _cache_size = 0;
-            _allocated_cache_size = 0;
-            _deallocated_cache_size = 0;
```
Comment (LiShuMing) [inline be/src/runtime/current_thread.h:69]: how to update `_allocated_cache_size ` & `_deallocated_cache_size`  if  `is_ctx_shift ` is false?

Comment (liuyehcf) [reply be/src/runtime/current_thread.h:69]: It's guaranteed by  RAII, the CurrentThreadMemTrackerSetter's destructor will invoke `commit(true)`

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Pull request merged**